src/components/Tree/TreeItem/TreeItemMultiselect.tsx

```diff
@@ -0,0 +1,192 @@
+
+        const toggleExpand = useCallback(
+            (event?: MouseEvent<HTMLButtonElement>) => {
+                event?.stopPropagation();
+                isExpanded ? onShrink?.(id) : onExpand?.(id);
+            },
+            [id, isExpanded, onExpand, onShrink],
+        );
+
+        useEffect(() => {
+            if (isParentSelected && !isSelected && onSelect) {
+                onSelect(id, true);
+            }
+        }, [id, onSelect, isParentSelected, isSelected, parentId]);
+
+        const hasChildren = Children.count(children) > 0;
+
+        const { enrichedChildren, childrenIds } = useMemo(() => {
+            const enrichedChildren = removeFragmentsAndEnrichChildren(children, {
+                parentId: id,
+                level: level + 1,
+            });
+
+            return {
+                enrichedChildren,
+                childrenIds: enrichedChildren.map((child) => child.props.id),
+            };
+        }, [children, id, level]);
+
+        useDeepCompareEffect(() => {
+            if (Children.count(enrichedChildren) === 0 || !isExpanded) {
+                unregisterNodeChildren?.(id);
+                return;
+            }
+
+            registerNodeChildren?.({ id, children: enrichedChildren });
+        }, [isExpanded, enrichedChildren, id]);
+
+        const itemStyleProps = useMemo(() => {
+            return {
+                spacingY: 'none',
+                contentHight: 'single-line',
+                shadow: 'none',
+                borderRadius: 'small',
+                borderWidth: 'none',
+                borderStyle: 'none',
+                activeColorStyle: 'neutral',
+                ...itemStyle,
+            } as TreeItemStyling;
+        }, [itemStyle]);
+
+        const { liClassName, backgroundClassName } = useMemo(() => {
+            return {
+                liClassName: getMultiselectLiClassName(itemStyleProps, isDisabled),
+                backgroundClassName: getMultiselectBackgroundClassName(itemStyleProps, isSelected, isDisabled),
+            };
+        }, [itemStyleProps, isDisabled, isSelected]);
+
+        const wrapperContentClassName = 'tw-max-w-full	tw-grow';
+        const showChildren = isExpanded;
+        const showLabel = label !== undefined;
+        const showExpandButton = expandable && (showCaret ?? hasChildren);
+
+        const containerClassName = getMultiselectContainerClassName(itemStyleProps);
+
+        const levelPadding = level * INDENTATION_WIDTH;
+        const liStyle = { paddingLeft: levelPadding };
+        const backgroundStyle = itemStyleProps.borderWidth !== 'none' ? {} : { marginLeft: -1 * levelPadding };
+        const style = {};
+
+        const checkBoxOnSelect = isDisabled ? () => void 0 : () => onSelect?.(id, false);
+        const checkBox =
+            checkBoxPosition !== 'none' ? (
+                <Container maxWidth="16px" maxHeight="16px">
```

Comment:
Is the `Container` necessary when the default size of the checkbox is 16px?

Reply:
It is needed with maxWidth, otherwise the checkbox wrapper expands... I can remove the unneeded `maxHeight`